enterprise/cmd/cody-gateway/internal/actor/actor.go

```diff
@@ -127,6 +127,7 @@ func (a *Actor) Limiter(
 	logger log.Logger,
 	redis limiter.RedisStore,
 	feature codygateway.Feature,
+	rateLimitAlerter func(actor *Actor, feature codygateway.Feature, usagePercentage float32, ttl time.Duration),
```

Comment:
Can we extract this into a type to avoid repeating it at all the various constructors?

```suggestion
	rateLimitAlertFunc RateLimitHitAlertFunc,
```

Reply:
We could also have some docstrings here, e.g. there are some nil checks at usage sites so maybe we can indicate this is optional

Reply:
Addressed in https://github.com/sourcegraph/sourcegraph/pull/53137/commits/91dda0a9c165dfea05ae8ab292795559f8b9f058, though probably just better to view the whole changeset since some changes are canceled out.